File: Chapter 03/Python/3.3.4 ISP/After/goalie.py
from interfaces import FieldPlayerTraining, GoalieTraining

class Goalie(FieldPlayerTraining, GoalieTraining):
    def practice_shooting(self):
        print("  [Goalie] Practicing goal kicks and long shots.")

    def practice_tackling(self):
        print("  [Goalie] Practicing 1-on-1 box tackles.")

    def practice_diving_saves(self):
        print("  [Goalie] Practicing top-corner diving saves.")

    def practice_hand_distribution(self):
        print("  [Goalie] Practicing fast break throws.")

File: Chapter 03/C#/3.3.4 ISP/After/Goalie.cs
using System;

namespace Chapter03.ISP.After;

// Goalies practice everything!
public class Goalie : IFieldPlayerTraining, IGoalieTraining
{
    public void PracticeShooting() 
    { 
        Console.WriteLine("  [Goalie] Practicing goal kicks and long shots."); 
    }
    
    public void PracticeTackling() 
    { 
        Console.WriteLine("  [Goalie] Practicing 1-on-1 box tackles."); 
    }
    
    public void PracticeDivingSaves() 
    { 
        Console.WriteLine("  [Goalie] Practicing top-corner diving saves."); 
    }
    
    public void PracticeHandDistribution() 
    { 
        Console.WriteLine("  [Goalie] Practicing fast break throws."); 
    }
}